internal/page_wasm.go

```diff
@@ -0,0 +1,8 @@
+//go:build wasm
+
+package internal
+
+func Getpagesize() int {
+	// A WebAssembly page has a constant size of 65,536 bytes, i.e., 64KiB
+	return 64 * 1024
```

Comment:
I don't know much about tinygo, but some Googling reveals that it implements a subset of the Go "os" package, and it looks like it implements os.Getpagesize() as well for the arches it targets. Why is it hardcoded here? This is the crux of the PR but isn't mentioned anywhere.

In any case, I'd like this to go into the respective stdlib, either in Google's implementation or tinygo if needed.

Reply:
I get the following error if I try to use os.Getpagesize:
```
$ tinygo build -o wasm.wasm main.go
# github.com/cilium/ebpf/internal
../../cilium/ebpf/internal/page_wasm.go:8:12: undefined: os.Getpagesize
```

According to https://pkg.go.dev/github.com/tinygo-org/tinygo/src/os#Getpagesize, it was added in tinygo 0.24.0 and I use 0.30.0, but it still does not work.

Issue: https://github.com/tinygo-org/tinygo/issues/1285

I agree it should ideally be fixed in tinygo or Go... I could try to make a PR there. But I would like to have a workaround in cilium/ebpf meanwhile. Btw, `syscall.Getpagesize()` works, presumably thanks to https://github.com/tinygo-org/tinygo/issues/2856; could that be an option to use that in cilium/ebpf as a workaround?

In https://github.com/tinygo-org/tinygo/issues/1285, they suggest to return -1... That would not work for cilium/ebpf when used in `unix.Mmap` but we wouldn't call those methods from wasm context. And for `internal.NewBufferedSectionReader(n=-1)`, it seems the implementation would use a buffer of 16 bytes when given a negative number, so that would work (but might be slow).

Reply:
> I could try to make a PR there. But I would like to have a workaround in cilium/ebpf meanwhile.

If we merge a workaround, there's no guarantee you will get around to making that upstream contribution. :wink: I really don't want to hardcode non-bpf-related platform constants, that's definitely the runtime/stdlib's responsibility.

> `syscall.Getpagesize()` works

Underlyingly, `os.Getpagesize()` simply calls `syscall.Getpagesize()` (https://cs.opensource.google/go/go/+/refs/tags/go1.23.0:src/os/types.go;l=13), so sounds like a trivial fix in tinygo? Might be a good exercise to understand the underlying tech a bit better.

I prefer to make only the absolute minimum changes to the lib to force us to address issues in the underlying layers. This is good for the health of the platform and potentially enables more software to run correctly out-of-the-box.